### PR TITLE
interfaces_file: re.escape() old value

### LINKED ISSUE #777: interfaces_file: Regular expression characters not escaped
<!--- Verify first that your issue is not already reported on GitHub -->
<!--- Also test if the latest release and devel branch are affected too -->
<!--- Complete *all* sections as described, this form is processed automatically -->

##### SUMMARY
<!--- Explain the problem briefly below -->
When replacing a value in /etc/network/interfaces that contains, for example, `||`, the new line is garbage. This is caused by the old value being used directly in a regular expression here:
https://github.com/ansible-collections/community.general/blob/4e56347fc1d79d8f0a1e8f663eca58ee17fb8e47/plugins/modules/system/interfaces_file.py#L287

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below, use your best guess if unsure -->
interfaces_file

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes -->
```paste below
ansible 2.9.10                                                                                                                                                                                                                                
  config file = /home/jpl/ansible/ansible.cfg
  configured module search path = ['/home/jpl/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/jpl/.local/lib/python3.6/site-packages/ansible
  executable location = /home/jpl/.local/bin/ansible
  python version = 3.6.9 (default, Jul 17 2020, 12:50:27) [GCC 8.4.0]
```

##### CONFIGURATION
<!--- Paste verbatim output from "ansible-config dump --only-changed" between quotes -->
```paste below
```

##### OS / ENVIRONMENT
<!--- Provide all relevant information below, e.g. target OS versions, network device firmware, etc. -->
Target Debian Buster, but probably irrelevant

##### STEPS TO REPRODUCE
<!--- Describe exactly how to reproduce the problem, using a minimal test-case -->

<!--- Paste example playbooks or commands between quotes below -->
```yaml
- hosts: all
  gather_facts: no
  tasks:
    - interfaces_file:
        iface: dummy0
        option: description
        value: 'Funny || name'

    - interfaces_file:
        iface: dummy0
        option: description
        value: 'Another || funny || name'
```

<!--- HINT: You can paste gist.github.com links for larger files -->

##### EXPECTED RESULTS
<!--- Describe what you expected to happen when running the steps above -->
/etc/network/interfaces:
```
iface dummy0 inet manual
    [...]
    description Another || funny || name
```

##### ACTUAL RESULTS
<!--- Describe what actually happened. If possible run with extra verbosity (-vvvv) -->

/etc/network/interfaces:
```
iface dummy0 inet manual
    [...]
    descriptionAnother || funny || name Funny || name
```

<!--- Paste verbatim command output between quotes -->
```paste below

```

### PR DESCRIPTION
**Pull request by jplitza**
##### SUMMARY

Fixes #777 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

interfaces_file

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: changelogs/fragments/777-interfaces_file-re-escape.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+bugfixes:
+- interfaces_file - escape regular expression characters in old value (https://github.com/ansible-collections/community.general/issues/777)
```
Comment (felixfontein) [inline changelogs/fragments/777-interfaces_file-re-escape.yml:3]:
```suggestion
- interfaces_file - escape regular expression characters in old value (https://github.com/ansible-collections/community.general/issues/777).
```

**Review by felixfontein — COMMENTED** [review]
Besides that, LGTM.

**Review by felixfontein — APPROVED** [review]

**Comment by mator** [comment]
works for me...
shipit

**Comment by felixfontein** [comment]
@jplitza thanks a lot for fixing this!
@mator thanks for reviewing!

Does anyone of you want to backport this to ansible/ansible's stable-2.9 branch (so it gets included in a 2.9.x release)? If not I can also do it.

**Comment by felixfontein** [comment]
I created the backport to ansible/ansible's stable-2.9: ansible/ansible#71766

**Pull request merged**